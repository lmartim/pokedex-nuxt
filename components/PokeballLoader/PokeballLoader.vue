<template>
  <div class="wrapper">
    <div class="pokeball" />
  </div>
</template>

<script>
export default {}
</script>

<style scoped>
.wrapper {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.pokeball {
  width: 60px;
  height: 60px;
  background-color: #fff;
  border-radius: 50%;
  position: relative;
  overflow: hidden;
  border: 3px solid;
  animation: frames 0.8s linear 0s infinite;
}

.pokeball:after {
  content: '';
  position: absolute;
  width: 60px;
  height: 30px;
  background-color: red;
  border-bottom: 4px solid;
  left: 0;
}

.pokeball:before {
  content: '';
  position: absolute;
  background-color: #fff;
  width: 10px;
  height: 10px;
  border: 4px solid;
  border-radius: 50%;
  bottom: 22px;
  right: 22px;
  z-index: 1;
}

/* AnimationFrames */
@keyframes frames {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
</style>
