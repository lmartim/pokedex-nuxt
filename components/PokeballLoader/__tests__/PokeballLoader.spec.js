import { shallowMount } from '@vue/test-utils'
import PokeballLoader from '@/components/PokeballLoader/PokeballLoader.vue'

describe('components/PokeballLoader', () => {
  describe('PokeballLoader is rendering', () => {
    const wrapper = shallowMount(PokeballLoader)

    it('Checking PokeballLoader snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
