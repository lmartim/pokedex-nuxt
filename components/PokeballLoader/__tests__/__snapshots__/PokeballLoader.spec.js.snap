// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PokeballLoader PokeballLoader is rendering Checking PokeballLoader snapshot 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
